tools/tsp-client/package.json

```diff
@@ -34,7 +34,8 @@
     "prettier": "^3.0.1",
     "rimraf": "^5.0.1",
     "ts-node": "^10.9.1",
-    "typescript": "^5.1.6"
+    "typescript": "^5.1.6",
+    "@typespec/compiler": "0.50.0"
```

Comment:
Who is responsible for updating this pinned version, and when?  The tool uses a floating peerDep at runtime so it doesn't impact users, but it does impact our build and test.  Ideally this would be updated whenever the version is updated in azure-rest-api-specs/main.